src/popup/modules/UserInterface.js

```diff
@@ -109,8 +109,15 @@ const refreshQrCode = throttle(() => {
         CommonMessages.hideError();
     }
 
-    QrCreator.setTextInternal(text);
-    QrCreator.generate();
+    try {
+        QrCreator.setTextInternal(text);
+        QrCreator.generate();
+    } catch (e) {
+        if (e === "Data too long") {
+            CommonMessages.showError("Cannot generate Qr Code, data size too big.");
```

Comment:
All error messages should be internationalized, so they can be translated/localized into multiple languages. See https://github.com/TinyWebEx/common/blob/master/CONTRIBUTING.md#translations and https://developer.mozilla.org/Add-ons/WebExtensions/Internationalization#Providing_localized_strings_in__locales

Also: _QR_ code… :smile: